bin/render-activity-stream-html.js

```diff
@@ -100,26 +258,101 @@ function main() {
   // process.argv are paths
   const args = require("minimist")(process.argv.slice(2), {
     alias: {
-      baseUrl: "b",
-      title: "t",
-      locale: "l",
-      debug: false
+      addonPath: "a",
+      baseUrl: "b"
     }
   });
 
-  const options = Object.assign({}, DEFAULT_OPTIONS, args || {});
-  const {html, state} = prerender(options.locale);
-  options.direction = state.App.textDirection;
+  const baseOptions = Object.assign({debug: false}, DEFAULT_OPTIONS, args || {});
+  const allStrings = require(`${baseOptions.addonPath}/data/locales.json`);
+  const addonPath = path.resolve(__dirname, baseOptions.addonPath);
+
+  const localizedFiles = new Map([
+    ["activity-stream-initial-state.js", ({state}) => templateJs(state)],
+    ["activity-stream-prerendered.html", ({html, options}) => templateHTML(options, html)],
+    ["activity-stream.html", ({options}) => templateHTML(options)]
+  ]);
 
-  fs.writeFileSync(path.join(BASE_FILE_PATH, "activity-stream.html"), templateHTML(options));
-  fs.writeFileSync(path.join(BASE_FILE_PATH, "activity-stream-debug.html"),
-    templateHTML(Object.assign({}, options, {debug: true})));
-  fs.writeFileSync(path.join(BASE_FILE_PATH, "activity-stream-prerendered.html"), templateHTML(options, html));
-  fs.writeFileSync(path.join(BASE_FILE_PATH, "activity-stream-prerendered-debug.html"),
-    templateHTML(Object.assign({}, options, {debug: true}), html));
-  fs.writeFileSync(path.join(BASE_FILE_PATH, "activity-stream-initial-state.js"), templateJs(state));
+  // Save default locale's strings to compare against other locales' strings
+  let defaultStrings;
+  const hasOnlyDefaultStrings = strings => defaultStrings &&
+    !Object.keys(strings).some(key => strings[key] !== defaultStrings[key]);
+
+  // Process the default locale first then all the ones from mozilla-central
+  const localizedLocales = [];
+  for (const locale of [DEFAULT_LOCALE, ...MOZILLA_CENTRAL_LOCALES]) {
+    const strings = getStrings(locale, allStrings);
+    if (hasOnlyDefaultStrings(strings)) {
+      console.log(`Skipping "${locale}" because it uses the same strings as ${DEFAULT_LOCALE}.`);
+      continue;
+    }
+
+    const {html, state} = prerender(locale, strings);
+    const options = Object.assign({}, baseOptions, {
+      direction: state.App.textDirection,
+      locale,
+      title: state.App.strings.newtab_page_title
+    });
+
+    // Put locale-specific files in their own directory
+    const basePath = path.join(addonPath, "prerendered", locale);
+    mkdir("-p", basePath);
+    for (const [file, templater] of localizedFiles) {
+      fs.writeFileSync(path.join(basePath, file), templater({html, options, state}));
+    }
+    console.log(`Done writing "${locale}" localized files to ${basePath}.`);
```

Comment:
Personally, I would do something like this for the output:
```
Writing localized files to /whatever/system-addon/prerendered
✔ en-US
✔ debug
✔ ach
✔ ar
...
Skipped the following locales because they had the same strings as en-US: af, ak, an, as, ...
Done.
```

You could maybe try something with `process.stdout.write` to get the locales on the same line, but yeah this obviously up to you

Reply:
Maybe.. I.. got.. carried.. away…

![screen shot 2017-10-20 at 2 04 56 pm](https://user-images.githubusercontent.com/438537/31841770-d39ab1ec-b59f-11e7-93d4-3c2103bf4345.png)